### PR TITLE
Fix handling of protected prefixes

### PR DESCRIPTION
**Pull request by jeswr**
Fixes https://github.com/rubensworks/jsonld-streaming-parser.js/issues/112
Fixes https://github.com/rubensworks/jsonld-streaming-parser.js/issues/117

Supercedes https://github.com/rubensworks/jsonld-context-parser.js/pull/61.

This has been tested against https://github.com/rubensworks/jsonld-streaming-parser.js. I have additional test cases which I will PR to that library once this patch is released.

### CODE REVIEW

File: lib/ContextParser.ts [hunk 1]
```diff
@@ -287,18 +290,10 @@ Tried mapping ${key} to ${JSON.stringify(keyValue)}`, ERROR_CODES.INVALID_KEYWOR
...
-          // If the simple term def was a prefix, explicitly mark the term as a prefix in the expanded term definition,
-          // because otherwise we loose this information due to JSON-LD interpreting prefixes differently
-          // in simple vs expanded term definitions.
-          if (isPrefix) {
-            contextAfter[key]['@prefix'] = true;
-            contextBefore[key]['@prefix'] = true; // Also on before, to make sure the next step still considers them ==
-          }
         }
 
+
```
Comment (jeswr) [inline lib/ContextParser.ts:296]:
```suggestion
```

File: test/ContextParser-test.ts [hunk 2]
```diff
@@ -2132,6 +2132,216 @@ Tried mapping @id to {}`, ERROR_CODES.KEYWORD_REDEFINITION));
         }));
       });
 
+      it('should parse 2 contexts where one is protected', () => {
+        return expect(parser.parse([
+          {
+            // NOTE: This gets elevated to 1.1; when the contexts get merged.
+            // We should add a test here which uses a 1.0 specific feature.
```
Comment (jeswr) [inline test/ContextParser-test.ts:2139]:
```suggestion
```

Captured by https://github.com/rubensworks/jsonld-context-parser.js/issues/66

**Comment by jeswr** [comment]
@rubensworks this is ready for review - I just don't have the permissions to assign you as a reviewer

File: test/ContextParser-test.ts [hunk 3]
```diff
@@ -2132,6 +2132,214 @@ Tried mapping @id to {}`, ERROR_CODES.KEYWORD_REDEFINITION));
...
+        "VerifiableCredential": {
+          "@id": "https://www.w3.org/2018/credentials#VerifiableCredential",
+        }
+      }
+
+      it('protected context should override the unprotected context', () => {
+        return expect(parser.parse([
+          {
+            // NOTE: This gets elevated to 1.1; when the contexts get merged.
+            // We should add a test here which uses a 1.0 specific feature.
```
Comment (jeswr) [inline test/ContextParser-test.ts:2205]:
```suggestion
```

**Comment by rubensworks** [comment]
Thanks! Released as 2.3.2.

**Pull request merged**